subsys/trusted_storage/src/aead/aead_ctr_nonce.c

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <psa/crypto.h>
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/util.h>
+
+#include <stdio.h>
+#include <stdint.h>
+#include <stddef.h>
+#include <string.h>
+#include <errno.h>
+
+#include "aead_nonce.h"
+
+/*
+ * NONCE is an incrementing 128bits number
+ */
+
+#define NONCE_MAX_LENGTH 16
+
+static uint64_t g_nonce_low;
```

Comment:
This can be a structure, so you can do one single call to generate random (using sizeof)

Reply:
Good idea, changed it